torch/_dynamo/logging.py

```diff
@@ -78,8 +83,24 @@ def init_logging(log_level, log_file_name=None):
 
 _step_counter = itertools.count(1)
 
+# Update num_steps if more phases are added: Dynamo, AOT, Backend
+# This is very inductor centric
+# _inductor.utils.has_triton() gives a circular import error here
+
+if not disable_progress:
+    try:
+        import triton  # noqa: F401
+
+        num_steps = 3
+    except ImportError:
+        num_steps = 2
+    pbar = tqdm(total=num_steps, desc="torch.compile()", delay=15)
```

Comment:
Is it possible to determine `num_steps` by looking at the final value of `_step_counter`?

Reply:
So the issue there is I need to know the final number of steps at the very beginning otherwise I'm not really showing a compilation progress bar but I'd just be logging events